### ISSUE TITLE
It is slow than pytorch-mobilenet-v1

### ISSUE THREAD

**Issue by guanfuchen** [tonylins/pytorch-mobilenet-v2#6]
I run the model on Mac, but it is slow than [pytorch-mobilenet-v1](https://github.com/marvis/pytorch-mobilenet/blob/master/main.py), ('MobileNetV2 forward time:', 0.29175710678100586), ('MobileNet forward time:', 0.2466120719909668), but I check the paper the arch is faster, does the different implementing result?
![image](https://user-images.githubusercontent.com/22321977/35388684-b909313e-020f-11e8-809e-6b77954337ee.png)

**Comment by windid** [comment]
Well, I've tried to implement MobilenetV2 in tensorflow, it's also slower than V1, evan though the Flops is lower.

And the MobilenetV2 implementation in tensorflow is much slower than pytorch.

I guess both of these problems are caused by separable conv.

**Comment by foreverYoungGitHub** [comment]
I meet the same problem in tensorflow. The paper use the TF-Lite and Pixel 1 to do the experiment, which may have some optimisation for separable conv. (v2 import more separate conv than v1).

**Comment by ouceduxzk** [comment]
@windid  you are right, but still using mobilenetv2 for transfer learning using part of the network is still possible to have some acceleration even though the currently tf impl is slow

**Comment by austingg** [comment]
@windid  have you ever freezed the graph and fuse the bn,  since mobilenet has more conv layers, and each conv has a batch normalization layer, and BN contrib some overhead.

**Comment by windid** [comment]
@austingg No, never try that. 
I think the speed bottleneck of this model is the memory bandwidth of the device.

Suppose K is the output channels of a conv layer:
For normal 3x3 conv, **every feature point** is computed by 3 * 3 * K times;
For 1x1 conv,  is computed by K times;
For separable 3x3 conv, is only computed by 3 * 3 = 9 times.

So the computation density is:
normal conv 3x3 > 1x1 conv  > separable conv

As mentioned by @foreverYoungGitHub , MobilenetV2 employs more separable conv, and the channels(K) on 1x1 conv are fewer, so this model is much more easier stuck at memory bandwidth, especially when you running this on GPU or CPU of PC.

Good news is on most mobile devices, the memory bandwidth was much  larger compared with its computation capacity. So when running on these devices, MobilenetV2 is probably faster than V1.

Sorry for my poor English, hope you can get it.

**Comment by austingg** [comment]
@windid  yes.  And the paper conduct the experiment on Google Pixel with tflite and **quantized(int8)**  ,which also consume less memory bandwidth.

**Comment by TangAL0203** [comment]
@windid good analysis!
